tests/templates/test_subroutines/test_qdrift.py

```diff
@@ -0,0 +1,592 @@
+        expected_state = tf.linalg.matvec(
+            reduce(
+                lambda x, y: x @ y,
+                [qml.matrix(op, wire_order=[0, 1]) for op in expected_decomp],
+            ),
+            initial_state,
+        )
+        state = circ(time)
+
+        assert allclose(expected_state, state)
+
+    @pytest.mark.jax
+    @pytest.mark.parametrize("seed", (1234, 42))
+    @pytest.mark.parametrize("coeffs, ops", test_hamiltonians)
+    def test_execution_jax(self, coeffs, ops, seed):
+        """Test that the circuit executes as expected using jax"""
+        from jax import numpy as jnp
+
+        time = jnp.array(0.5)
+        dev = qml.device("default.qubit", wires=[0, 1])
+
+        @qml.qnode(dev)
+        def circ(time):
+            hamiltonian = qml.dot(coeffs, ops)
+            qml.QDrift(hamiltonian, time, n=2, seed=seed)
+            return qml.state()
+
+        expected_decomp = _sample_decomposition(coeffs, ops, time, n=2, seed=seed)
+
+        initial_state = jnp.array([1.0, 0.0, 0.0, 0.0])
+
+        expected_state = (
+            reduce(
+                lambda x, y: x @ y,
+                [qml.matrix(op, wire_order=[0, 1]) for op in expected_decomp],
+            )
+            @ initial_state
+        )
+        state = circ(time)
+
+        assert allclose(expected_state, state)
+
+    @pytest.mark.jax
+    @pytest.mark.parametrize("seed", (1234, 42))
+    @pytest.mark.parametrize("coeffs, ops", test_hamiltonians)
+    def test_execution_jaxjit(self, coeffs, ops, seed):
+        """Test that the circuit executes as expected using jax jit"""
+        import jax
+        from jax import numpy as jnp
+
+        time = jnp.array(0.5)
+        dev = qml.device("default.qubit", wires=[0, 1])
+
+        @jax.jit
+        @qml.qnode(dev, interface="jax")
+        def circ(time):
+            hamiltonian = qml.sum(*(qml.s_prod(coeff, op) for coeff, op in zip(coeffs, ops)))
+            qml.QDrift(hamiltonian, time, n=2, seed=seed)
+            return qml.state()
+
+        expected_decomp = _sample_decomposition(coeffs, ops, time, n=2, seed=seed)
+
+        initial_state = jnp.array([1.0, 0.0, 0.0, 0.0])
+
+        expected_state = (
+            reduce(
+                lambda x, y: x @ y,
+                [qml.matrix(op, wire_order=[0, 1]) for op in expected_decomp],
+            )
+            @ initial_state
+        )
+        state = circ(time)
+
+        assert allclose(expected_state, state)
+
+    @pytest.mark.all_interfaces
+    @pytest.mark.parametrize(
+        "interface",
+        (
+            # "tf",
+            # "jax",
+            # "torch",
+            "autograd",
+        ),
+    )
+    def test_error_gradient_workflow(self, interface):
+        """Test that an error is raised if we require a gradient of QDrift with respect to hamiltonian coefficients."""
+        if interface == "autograd":
+            time = qnp.array(1.5)
+            coeffs = qnp.array([1.23, -0.45], requires_grad=True)
```

Comment:
Do we need this?

Reply:
Yes, to check that the error is raised correctly when trying to take gradient with respect to coefficients